File: backend/src/main/java/com/sentinel/backend/entity/PermissaoGrupo.java
package com.sentinel.backend.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "permissoes_grupo")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PermissaoGrupo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome; // Ex: ADMIN, TECNICO

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "grupo_paginas", joinColumns = @JoinColumn(name = "grupo_id"), inverseJoinColumns = @JoinColumn(name = "pagina_id"))
    private List<PermissaoPagina> permissoes;
}

File: frontend/src/app/models/permissao-grupo.ts
import { PermissaoPagina } from './permissao-pagina';

export class PermissaoGrupo {
  id?: number;
  nome: string;
  permissoes: PermissaoPagina[];

  constructor(nome: string, id?: number, permissoes: PermissaoPagina[] = []) {
    this.id = id;
    this.nome = nome;
    this.permissoes = permissoes;
  }
}
